Content.Server/Speech/EntitySystems/DamagedSiliconAccentSystem.cs

```diff
@@ -0,0 +1,151 @@
+
+        var idx = -1;
+        foreach (var letter in message)
+        {
+            idx++;
+            if (idx < idxMin) // Fast character, no effect
+            {
+                outMsg.Append(letter);
+                continue;
+            }
+
+            // use an x^2 interpolation to increase the drop probability until we hit idxMax
+            var probToDrop = idx >= idxMax
+                ? maxDropProb
+                : (float)Math.Pow(((double)idx - idxMin) / (idxMax - idxMin), 2.0) * maxDropProb;
+            // Ensure we're in the range for Prob()
+            probToDrop = Math.Clamp(probToDrop, 0.0f, 1.0f);
+
+            if (_random.Prob(probToDrop)) // Lose a character
+            {
+                // Additional chance to change to dot for flavor instead of full drop
+                if (_random.Prob(probToReplaceWithDot))
+                {
+                    outMsg.Append('.');
+                }
+            }
+            else // Character is safe
+            {
+                outMsg.Append(letter);
+            }
+        }
+        return outMsg.ToString();
+    }
+
+    public string CorruptDamage(string message, FixedPoint2 totalDamage)
+    {
+        var outMsg = new StringBuilder();
+        // Linear interpolation of character damage probability
+        var damagePercent = Math.Clamp((float)totalDamage / DamageAtMaxCorruption, 0, 1);
+        var chanceToCorruptLetter = damagePercent * MaxCorruption;
+        foreach (var letter in message)
+        {
+            if (_random.Prob(chanceToCorruptLetter)) // Corrupt!
+            {
+                outMsg.Append(CorruptLetterDamage(letter));
+            }
+            else // Safe!
+            {
+                outMsg.Append(letter);
+            }
+        }
+        return outMsg.ToString();
+    }
+
+    private string CorruptLetterDamage(char letter)
+    {
+        var res = _random.NextDouble();
+        return res switch
+        {
+            < 0.0 => letter.ToString(), // shouldn't be less than 0!
+            < 0.5 => CorruptPunctuize(), // 50% chance to replace with random punctuation
+            < 0.75 => "", // 25% chance to remove character
+            < 1.00 => CorruptRepeat(letter), // 25% to repeat the character
+            _ => letter.ToString(), // shouldn't be greater than 1!
+        };
+    }
+
+    private string CorruptPunctuize()
+    {
+        const string punctuation = "\"\\`~!@#$%^&*()_+-={}[]|\\;:<>,.?/";
+        return punctuation[_random.NextByte((byte)punctuation.Length)].ToString();
+    }
+
+    private string CorruptRepeat(char letter)
+    {
+        // If we got here that means we're at least repeating it once
+        var res = new StringBuilder($"{letter}{letter}");
+        // 25% chance to add another character in the streak
+        // (kind of like "exploding dice")
+        while (_random.Prob(0.25f))
+        {
+            res.Append(letter);
+        }
```

Comment:
Doing randoms in a while loop is a bad idea due to RNG.

Reply:
I was thinking about that, in case there is an implementation of Robust.Random that might not actually put out random numbers....

I'll just numerically calculate up to 5 repeats and hard code those in...

Reply:
removed the RNG based loop